taskfile/read/node_http.go

```diff
@@ -0,0 +1,56 @@
+package read
+
+import (
+	"net/http"
+	"net/url"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/go-task/task/v3/errors"
+	"github.com/go-task/task/v3/taskfile"
+)
+
+// An HTTPNode is a node that reads a Taskfile from a remote location via HTTP.
+type HTTPNode struct {
+	BaseNode
+	URL *url.URL
+}
+
+func NewHTTPNode(includedTaskfile taskfile.IncludedTaskfile, parent Node) (*HTTPNode, error) {
+	url, err := url.Parse(includedTaskfile.Taskfile)
+	if err != nil {
+		return nil, err
+	}
+
+	return &HTTPNode{
+		BaseNode: BaseNode{
+			parent:   parent,
+			optional: includedTaskfile.Optional,
+		},
+		URL: url,
+	}, nil
+}
+
+func (node *HTTPNode) Location() string {
+	return node.URL.String()
+}
+
+func (node *HTTPNode) Read() (*taskfile.Taskfile, error) {
+	resp, err := http.Get(node.URL.String())
+	if err != nil {
+		return nil, errors.TaskfileNotFoundError{URI: node.URL.String()}
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.TaskfileNotFoundError{URI: node.URL.String()}
+	}
```

Comment:
Perhaps consider adding more information to the error? (`err.Error()` or the status code).

Otherwise, if the request fails, the user will have a hard time debugging it.